### PR TITLE
feat: support notation cli end-to-end test framework

### PR DESCRIPTION
**Pull request by chloeyin**
## What
Suppor end-to-end test by using [Ginkgo](https://github.com/onsi/ginkgo)
Add e2e test infrastructure for notation-cli
The core functions are in **test/e2e/scenario** and **test/e2e/internal/utils/exec.go**
There is a quick introduction in **test/e2e/Readme.md**

- Closes https://github.com/notaryproject/roadmap/issues/47

Signed-off-by: zaihaoyin <zaihaoyin@microsoft.com>

### CODE REVIEW

File: test/e2e/Readme.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+1. Github Actions
+
+    * Please check `Run e2e tests` steps in **workflows/build.yml** for detail.
+2. Local environment
+
+    * Install Golang.
+    * Install Docker.
+    * install Ginkgo CLI.
+    * Clone the repository.
+    * Run `./test/e2e/local.sh absolute_path_to_your_notation_binary` in the repository's root directory. It is a simple shell script which only sets some environment variables, builds the notation image and starts the test registry.
```
Comment (JeyJeyGao) [inline test/e2e/Readme.md:48]: Is it easier to use by adding a Makefile `test-e2e` command to build notation and call this script together.

File: test/e2e/testdata/config/config-example-with-extensions.yml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    X-Content-Type-Options: [nosniff]
+health:
+  storagedriver:
+    enabled: true
+    interval: 10s
+    threshold: 3
+auth:
+  htpasswd:
+    realm: test-basic
+    path: /etc/docker/registry/passwd # hello:oras-test
```
Comment (qweeah) [inline test/e2e/testdata/config/config-example-with-extensions.yml:25]: And the bcrpyt password need to be changed too.
```suggestion
    path: /etc/docker/registry/passwd # hello:nonation-test
```

**Review by jondonas — APPROVED** [review]

**Comment by dtzar** [comment]
@yizha1 - can we just fix the go.mod/sum conflicts and merge and iterate from there?

**Comment by yizha1** [comment]
@dtzar The test framework needs to be updated based on the ORAS E2E test framework. We may not get it done in rc.1 release @vaninrao10

**Comment by yizha1** [comment]
Move to rc.2 since test framework need to be improved based on the practice in ORAS project.

**Comment by chloeyin** [comment]
Close this PR as we have a new [framework](https://github.com/notaryproject/notation/pull/493) based on this PR

**Pull request closed without merging**